### PR TITLE
Enforce unique names across collections, and repository roles

### PR DESCRIPTION
**Pull request by awcrosby**
- On collection import, check for name conflict with repository roles
  - the check occurs in the API collection list view post.
- On repository import, check for name conflict with collections
  - the check occurs in the worker, which is called from multiple API endpoints (repository, notifications, import task). The check occurs after the importer has processed the repository and potentially renamed the repository based on the `meta/main.yml` `role_name` attribute

### CODE REVIEW

File: galaxy/api/v2/views/collection.py [hunk 1]
```diff
@@ -143,6 +150,21 @@ def _check_namespace_access(self, namespace, user):
                 'the namespaces you have access to.'
             )
 
+    def _check_role_name_conflict(self, ns, name):
+        roles = models.Content.objects.filter(
+            content_type__name=constants.ContentType.ROLE,
+            namespace=ns,
```
Comment (awcrosby) [inline galaxy/api/v2/views/collection.py]: Options to consider here:
- `namespace=ns`
- `repository__provider_namespace__namespace=ns`

**Comment by ironfroggy** [comment]
galaxy-duplicate

I was still able to create a duplicate by publishing a collection named "just_a_role" and then importing my `orionuser1.just-a-role` repository.

In the other direction, it worked fine. I wasn't able to publish the collection if I imported the repository first.

File: galaxy/api/v2/views/collection.py (hunk 1 above)
Comment (chouseknecht) [reply galaxy/api/v2/views/collection.py]: We have `namespace` on the Content object for convenience, but I think to be accurate, we should use `provider_namespace__namespace`.

**Comment by chouseknecht** [comment]
Following up on comment from @ironfroggy, this looks like a case of converting `-` to `_`. We need to account for that in the duplicate check.

**Comment by awcrosby** [comment]
@ironfroggy can you confirm this occurs after you switch to this branch and restart the celery workers? This scenario should be taken care of

> I was still able to create a duplicate by publishing a collection named "just_a_role" and then importing my orionuser1.just-a-role repository.

File: galaxy/api/v2/views/collection.py (hunk 1 above)
Comment (awcrosby) [reply galaxy/api/v2/views/collection.py]: update was made to `repository__provider_namespace__namespace`

**Pull request merged**